plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java

```diff
@@ -320,11 +323,25 @@ public void testEnum()
     public void testUuid()
     {
         SqlDataTypeTest.create()
-                // TODO map to Trino UUID
-                .addRoundTrip("UUID", "'417ddc5d-e556-4d27-95dd-a34d84e46a50'", createUnboundedVarcharType(), "CAST('417ddc5d-e556-4d27-95dd-a34d84e46a50' AS varchar)")
-                .execute(getQueryRunner(), clickhouseCreateAndInsert("tpch.test_uuid"));
+                .addRoundTrip("UUID", "'114514ea-0601-1981-1142-e9b55b0abd6d'", UuidType.UUID, "cast('114514ea-0601-1981-1142-e9b55b0abd6d' as UUID)")
+                .execute(getQueryRunner(), clickhouseCreateAndInsert("default.ck_test_uuid"));
 
-        // TODO add test with UUID written from Trino
+        SqlDataTypeTest.create()
+                .addRoundTrip("uuid", "UUID '114514ea-0601-1981-1142-e9b55b0abd6d'", UuidType.UUID, "cast('114514ea-0601-1981-1142-e9b55b0abd6d' as uuid)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("default.ck_test_uuid"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("uuid", "UUID '114514ea-0601-1981-1142-e9b55b0abd6d'", UuidType.UUID, "cast('114514ea-0601-1981-1142-e9b55b0abd6d' as uuid)")
+                .execute(getQueryRunner(), trinoCreateAndInsert("default.ck_test_uuid"));
+
+        // test null value
+        String tableName = "test_uuid_null" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(x uuid)");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+        assertUpdate("INSERT INTO " + tableName + " VALUES(null)", 1);
+        assertQuery("SELECT COUNT(*) FROM " + tableName, " VALUES (1)");
+        assertQuery("SELECT COUNT(*) FROM " + tableName + " WHERE x IS null", " VALUES (1)");
+        assertUpdate("DROP TABLE " + tableName);
```

Comment:
You can simply do this instead. If the types are explicitly provided in the `addRoundTrip` then they are used instead of doing a CREATE TABLE AS SELECT.

```java
@Test
public void testUuid()
{
    SqlDataTypeTest.create()
            .addRoundTrip("Nullable(UUID)", "NULL", UuidType.UUID, "CAST(NULL AS UUID)")
            .addRoundTrip("Nullable(UUID)", "'114514ea-0601-1981-1142-e9b55b0abd6d'", UuidType.UUID, "CAST('114514ea-0601-1981-1142-e9b55b0abd6d' AS UUID)")
            .execute(getQueryRunner(), clickhouseCreateAndInsert("default.ck_test_uuid"));
                                                                                                                                                            
    SqlDataTypeTest.create()
            .addRoundTrip("CAST(NULL AS UUID)", "cast(NULL as UUID)")
            .addRoundTrip("UUID '114514ea-0601-1981-1142-e9b55b0abd6d'", "CAST('114514ea-0601-1981-1142-e9b55b0abd6d' AS UUID)")
            .execute(getQueryRunner(), trinoCreateAsSelect("default.ck_test_uuid"))
            .execute(getQueryRunner(), trinoCreateAndInsert("default.ck_test_uuid"));
}
```

Reply:
got it